rancid-tomatillos/src/MoviePage/MoviePage.js

```diff
@@ -1,8 +1,42 @@
-import React from 'react'
+import React from "react";
+import api from '../api';
 
-function ShowPage(props) {
-    return <h1>isworking</h1>
-}
+function MoviePage(props) {
+  let id = props.location.pathname.split("/")[2];
 
+  if (props.match.isExact) {
+    console.log(props);
+    const {
+      average_rating,
+      backdrop_path,
+      budget,
+      genres,
+      id,
+      overview,
+      poster_path,
+      release_date,
+      revenue,
+      runtime,
+      tagline,
+      title,
+    } = props.location.state.movie;
+    return (
+      <div itemID={id} onMouseOver={console.log(overview)}>
+        <h1 className="movie-title">{title}</h1>
+        <img src={backdrop_path} alt={`${title} background`} />
+        <p className="average-rating"> Average user rating: {average_rating}</p>
+        <p className="genres"> {genres}</p>
+        <p className="release-date"> {release_date}</p>
+        <p className="revenue">{revenue ? revenue : "Not released"}</p>
+        <p className="runtime"> {runtime} minutes</p>
+        {budget ? <p className="budget">{budget}</p> : null}
+        {tagline ? <p className="tagline"> {tagline}</p> : null}
+      </div>
+    );
+  } else {
+    api.getIndividualMovie(id, props);
+    return <h1>Loading</h1>;
```

Comment:
@khalidwilliams 
It's a little hard to tell what is going on here, but we went a route for our individual movie page that seems a little awkward.  

1. `onClick` of any `previewCard` `<div> ` in the dashboard, `history.push(/movie${id})` is invoked
2. redirect to /movie${id} which triggers the else conditional since the url and location do not exact match (this invokes our single movie fetch)
3. once the fetch promise is resolved we invoke `history.push(path=/movie), (state=resolved promise)` which redirects to /movie
4. our if conditional is now true which renders the moviePage component with all the info we fetched

Wondering if we should be focusing more on utilizing setState to cause renders here?

Reply:
Wondering if this will be difficult to test as well? and if it makes more sense to create another class component that has state (or add onto another component with state already), or take advantage of quicker load with functional components?

Reply:
This is a bit convoluted. You might want to look into the Link from Router. We'll also look over it tomorrow during class, and it may give you some insight :) @hannakim91 @BlakeDonn